### PR TITLE
Elasticlunr

### PR DESCRIPTION
**Pull request by vinyll**
- this creates a JSON instead of an HTML. Similar to what we had before but with the playlist untouched.
- The JS for templating is very much simplified and generic compared to the "old" JS we had initially.
- This would permit to load extra cards [from the API](https://pgsearch.dokku.gouv.studio/api?q=searchme)
- search is based on [ElasticLunr](http://elasticlunr.com) sorting by relevancy.

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -248,16 +67,14 @@ def populate_excerpt(self, html_description: str, num_words: int = 50) -> None:
             truncated_description = Truncator(truncated_description).chars(
                 num=num_words * 10, truncate="…", html=True
             )
-        unlinkified_description = re.sub(r"http\S+", "", truncated_description)
-        self.excerpt = unlinkified_description
+        self.excerpt = re.sub(r"http\S+", "", truncated_description)
```
Comment (davidbgk) [inline run.py]: If you do that simplification, can you add a comment to explain what is being done?

**Comment by vinyll** [comment]
unminified elasticlunr was added in /js and a freshly generated datasets.json is in place.

File: run.py [hunk 2]
```diff
@@ -336,9 +149,8 @@ def deduplicate_datasets(datasets: List[Dataset]) -> List[Dataset]:
 
 
 def write_datasets(datasets: List[Dataset]) -> None:
-    template = environment.get_template("template.html")
-    content = template.render(datasets=[dataset.asdict for dataset in datasets])
-    open("index.html", "w").write(content)
+    data = [d.asdict for d in datasets]
+    open("datasets.json", "w").write(json.dumps(data))
```
Comment (davidbgk) [inline run.py]: Would it be possible to add the `indent=2` option to `json.dumps`? I try to estimate the number of lines of code we add with that PR.

Comment (vinyll) [reply run.py]: I've uploaded the datasets.json file in a formatted file with indents and unoptimized newlines ;)

**Comment by davidbgk** [comment]
There are a few regressions:

* no more fuzzy search `openfood` doesn't return any result anymore
* no more search terms stored within the URL
* do we still have pertinent statistics? (on typing) I don't see where the line `_paq.push(['trackEvent', 'Search', 'Type', searchTerms])` has been put.
* no more pre-filled terms on redirection to datagouv search

I should have reviewed it more carefully, sorry :/

**Pull request merged**